Sources/SpeziOnboarding/OnboardingFlow/OnboardingViewController.swift

```diff
@@ -0,0 +1,40 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import SwiftUI
+
+public class OnboardingViewController: ObservableObject {
+    var views: [AnyView]
+    var complete: Binding<Bool>
+    
+    @Published var currentView: AnyView
+    
+    init(views: [any View], complete: Binding<Bool>) {
+        self.views = views.map { AnyView($0) }  // Convert to `AnyView` (TODO?)
+        self.complete = complete
+        self.currentView = self.views.popLast() ?? .init(EmptyView())
+    }
+    
+    public func nextStep() {
```

Comment:
I would highly recommend that his method uses some internal state to keep track of the step (maybe the custom type defined in the comment above) and then just appends this to the NavigationPath. 
Letting a `NavigationStack` therefore handle all the complicated logic might sound like the best solution to me.